src/core/exception.d

```diff
@@ -579,6 +579,12 @@ extern (C) void onSwitchError( string file = __FILE__, size_t line = __LINE__ )
     throw new SwitchError( file, line, null );
 }
 
+// Compiler lowers final switch default case to this (which is a runtime error)
+void __switch_errorT()(string file = __FILE__, size_t line = __LINE__) @trusted
+{
+    // Consider making this a compile time check.
```

Comment:
We already have that compile time check for `final switch` and normal `switch` always requires a `default` branch.

I guess this runtime check is only used to verify that a value is not out-of-range, probably caused by an invalid `cast`. Ping @andralex

Reply:
@jpf91 take a look at https://dpaste.dzfl.pl/a33b4b74d774

Reply:
Thanks. I didn't know `final switch` also works on non-enumeral values.